src/app_service/service/message/service.nim

```diff
@@ -115,6 +120,13 @@ QtObject:
         if(chatType == ChatType.Unknown):
           error "error: new message with an unknown chat type received for chat id ", chatId
           return
+
+        # Handling edited message update
+        if(receivedData.messages.len == 1 and receivedData.messages[0].editedAt != 0):
```

Comment:
@Khushboo-dev-cpp this is good, but let's just for the sake of readability update this `init` proc, since it gets a bigger and bigger in a this way:
```
proc init*(self: Service) =
    self.events.on(SignalType.Message.event) do(e: Args):
      var receivedData = MessageSignal(e)

      if (receivedData.messages.len > 0 and receivedData.chats.len > 0):
        self.handleMessagesUpdate(receivedData.messages)
      if (receivedData.pinnedMessages.len > 0):
        self.handlePinnedMessagesUpdate(receivedData.pinnedMessages)
      if (receivedData.deletedMessages.len > 0):
        self.handleDeleteMessagesUpdate(receivedData.deletedMessages)
```

and just move corresponding code to that private procs which you're going to add.

Reply:
done